tot/models.py

```diff
@@ -225,3 +231,302 @@ def maybe_drop_added_dates(self, predicted, df):
             predicted_new, received_ID_col_pred, received_single_time_series_pred, received_dict_test_pred
         )
         return predicted, df
+
+
+@dataclass
+class SeasonalNaiveModel(Model):
+    """
+    A `SeasonalNaiveModel` is a naive model that forecasts future values of a target series based on past observations
+    of the target series of the specified period, i.e season.
+
+    The naive models act on time series data having ``fit()`` and ``predict()`` methods. The season can be specified
+    by K steps. Therefore, the model predicts the value of `K` time steps ago. When `K>1`, it repeats the last `K`
+    values.
+
+    Parameters
+    ----------
+        K : int
+            the number of last time steps to repeat
+        n_forecasts : int
+            Number of steps ahead of prediction time step to forecast.
+    Note
+    ----
+        ``Supported capabilities``
+        * univariate time series
+        * single target
+        * manually select season steps K
+        * n_forecats>1
+
+        ``Not supported capabilities``
+        * autoregression
+        * past covariates
+        * future covariates
+
+        ``Planned capabilities``
+        * auto-detect season
+        * multi-variate time series only with multi-target
+        * frequency check and optional frequency conversion
+    """
+
+    model_name: str = "SeasonalNaive"
+    model_class: Type = None  # make sense?
+
+    def __post_init__(self):
+        # no installation checks required
+
+        # expect no _data_params
+        data_params = self.params["_data_params"]
+        assert len(data_params) == 0
+        custom_seasonalities = None  # TODO: auto-detect season from metadata
```

Comment:
Ideally we just get K from the set of seasonalities defined in the dataset (and choose the smallest)?

Reply:
nice idea. I implemented it that way. First, we check if the use wants to input a custom seasonality. If that is not the case, we assign K (now called season_length) to the smallest seasonality. In case the dataset doesn´t provide a dataset (and there is no user input) we throw an error message.